### PR TITLE
docs: editorial love for IBC overview

### PR DESCRIPTION
**Pull request by barriebyron**
Apply technical style standards, present tense, fix typos, and use consistent IBC language

## Description

We all know that IBC is how blockchains speak to each other. The world is watching, let's use our best editorial content quality standards for this essential content. 

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:30]: I wasn't sure what is optional, the acknowledgement structs or defining them

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:59]: I changed all instances of may to can
per Google developer documentation style guide advice for using can https://developers.google.com/style/word-list#letter-c 
and  may https://developers.google.com/style/word-list#may

key point

> may
> In general, reserve _may_ for official policy or legal considerations. To convey possibility, use _can_ or _might_ instead. To convey permission, use _can_ instead. See also can, might, and should.

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:65]: I applied the guideline to use present tense, per https://developers.google.com/style/tense

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:79]: for "This" I think we mean "This module action authentication"

beware the naked "This" especially for complex sentences

> this, that
> Where possible, put a noun after this or that for clarity. If doing so results in clunky prose, then don't do it; but even then, try thinking about what the noun would be. If you aren't sure what noun this or that refers to, then consider rephrasing.

see https://developers.google.com/style/word-list#letter-t

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:82]: another unclear reference "This"
I guessed that we are talking about "This dynamic capability store" 

But please teach me if I'm off track

**Review by barriebyron — COMMENTED** [review]
some explanation for my changes, teach me where I'm off track, I aim to help IBC content be the best content

**Comment by crodriguezvega** [comment]
Thanks @barriebyron for giving some ❤️ to our documentation!

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:23]:
```suggestion
protocol. The channel and port abstraction layer is most relevant to application developers, here we will define custom packets and
```

The second sentence didn't make sense to me.

File: docs/ibc/overview.md
Comment (AdityaSripal) [reply docs/ibc/overview.md:30]: You can define your own custom acknowledgement struct if you wish, or you can use the default one provided by core IBC.

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:31]:
```suggestion
- Standardize an encoding of the packet data
```

Comment (AdityaSripal) [reply docs/ibc/overview.md:31]: or you could say `Define` to keep it consistent

Comment (AdityaSripal) [reply docs/ibc/overview.md:31]: Though this point seems like it should be rolled into `Defining packet data struct`.

Perhaps `Define packet data struct and encoding`?

The same is true for acknowledgment

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:79]:
```suggestion
appropriate permissions can use them. This module authentication is accomplished using a [dynamic
```

Seems like `module authentication` is more appropriate

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:82]:
```suggestion
order to use that port or channel. The dynamic capability module prevents other modules from using that port or channel since
```

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:123]:
```suggestion
Modules can choose which channels they wish to communicate over, thus IBC expects modules to
```

**Review by AdityaSripal — CHANGES_REQUESTED** [review]
Thank you for bringing this doc more in line with technical writing standards! I have a few comments here that we should resolve before merging. If you disagree with any suggestion, please comment on them and we can resolve it there.

Thanks again!

File: docs/ibc/overview.md
Comment (damiannolan) [reply docs/ibc/overview.md:59]: Should this not read as: "Connections, once established, are responsible for **facilitating** all cross-chain verification of IBC state"?

Another small point which I am unsure of but I assumed previously - I thought the use of `may` here was as per [RFC-2119](https://datatracker.ietf.org/doc/html/rfc2119). @AdityaSripal maybe can you confirm?

Comment (AdityaSripal) [reply docs/ibc/overview.md:59]: > Connections, once established, are responsible for facilitating all cross-chain verification of IBC state

Yea that's right.

`MAY` implied optional as mentioned in the RFC, but i don't believe we stuck to a very strict technical writing standard so I defer to experts on the best way to convey this.

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:30]:
```suggestion
- Use the default acknowledgment struct provided by core IBC, or optionally define your own custom acknowledgment struct
```

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:32]: @crodriguezvega in this case, we can apply the Google developer documentation technical content style standards and add a period (.) at the end of each list item since the items include a verb.

My preference is to omit punctuation at the end of sentence fragments in lists; the key point is to be consistent throughout the content set. 

https://developers.google.com/style/periods?hl=en#bulleted-lists

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:59]: I'll use the gerund verb facilitating as recommended, good catch @damiannolan but then we need plural "verifications"

**Comment by barriebyron** [comment]
hi @AdityaSripal and @crodriguezvega thanks for the feedback and attention.
IBC docs are so essential to building the internet of blockchains... 

I have added a few more in-line questions and made these changes:

- used consistent spelling for acknowledgments 
- fixed some misplaced commas
- tightened up some loose sentences
- applied suggestion to add punctuation to end of list items
- typo fixes that I didn't catch on first round

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:186]: The unclear reference (the notorious naked "This") here makes me wonder what the subject is (I couldn't even guess... "This logic"? or "This relay"?)

please r/w or provide the subject of "This" in  

"This can involve rolling"

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:182]:
```suggestion
Regardless, this acknowledgment data is opaque to IBC much like the packet `Data` and is treated by IBC as a simple byte string `[]byte`. It is incumbent on receiver modules to encode their acknowledgment so that the sender module can decode it correctly. Define this encoding through version negotiation during the channel handshake.
```

Comment (barriebyron) [reply docs/ibc/overview.md:182]: I probably guessed incorrectly but again the unclear reference "This" in "This should be decided"

are we talking about the encoding? or ? and how is it decided?

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:174]: I didn't tackle this grammatical challenge, but now that we are all focused on clarity and overall improvements, let's see if we can't improve here 

```suggestion
- In UNORDERED channels, packets can be received in any order. Thus, IBC writes a packet receipt for each sequence it has received in the UNORDERED channel. This receipt contains no information; it is simply a marker intended to signify that the UNORDERED channel has received a packet at the specified sequence. To timeout a packet on an UNORDERED channel, proof is required that a packet receipt does not exist for the packet's sequence by the specified timeout. Of course, timing out a packet on an UNORDERED channel simply triggers the application-specific timeout logic for that packet and does not close the channel.
```

**Review by barriebyron — COMMENTED** [review]
take it from here, ya'll!

Thanks for the editorial partnering for awesomeness

File: docs/ibc/overview.md
Comment (crodriguezvega) [reply docs/ibc/overview.md:186]: I think it refers to the execution of the acknowledgment logic. So maybe something like `This acknowledgment logic can involve rolling...`.

There is also an extra space in `which  then`.

File: docs/ibc/overview.md
Comment (crodriguezvega) [reply docs/ibc/overview.md:182]: I think we're talking about how it is decided the encoding, so that both parties know how to encode/decode to understand each other.

File: docs/ibc/overview.md
Comment (AdityaSripal) [reply docs/ibc/overview.md:186]: Maybe we should replace that sentence with:

`Application developers can roll back packet-send changes in the case of a failed acknowledgement (e.g. refunding senders in ICS20).`

File: docs/ibc/overview.md
Comment (AdityaSripal) [reply docs/ibc/overview.md:182]: `The encoding should be negotiated between the two parties during version negotiation in the channel handshake.

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:174]:
```suggestion
- In UNORDERED channels, packets can be received in any order. Thus, IBC writes a packet receipt for each sequence it has received in the UNORDERED channel. This receipt contains no information; it is simply a marker intended to signify that the UNORDERED channel has received a packet at the specified sequence. In order to timeout a packet on an UNORDERED channel, a proof is required that a packet receipt **does not exist** for the packet's sequence by the specified timeout deadline. Timing out a packet on an UNORDERED channel prevents any further receiving of the packet, triggers the application-specific timeout logic for that packet, but does not close the channel.
```

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:182]:
```suggestion
Regardless, this acknowledgment data is opaque to IBC much like the packet `Data` and is treated by IBC as a simple byte string `[]byte`. It is incumbent on receiver modules to encode their acknowledgment so that the sender module can decode it correctly. The encoding should be negotiated between the two parties during version negotiation in the channel handshake.
```

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:186]:
```suggestion
After the acknowledgment has been written by the receiving chain, a relayer relays the acknowledgment back to the original sender module which  then executes application-specific acknowledgment logic using the contents of the acknowledgment. Application developers can roll back packet-send changes in the case of a failed acknowledgement (e.g. refunding senders in ICS20).
```

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:174]: see how the rewrite is now... let's try to write concisely what is happening. Instead of "deadline" let's use "timeout" (since it's timeout height, not a date-based deadline)

let's say what happens, not what doesn't happen

Very challenging! I feel for our users

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:182]: since this is the overview topic, I suggest that we are too specific here.

Readers can learn more at https://docs.cosmos.network/master/ibc/custom.html#channel-handshake-version-negotiation 
---
oh "should" is such an ambiguous word, let's avoid per this advice
https://developers.google.com/style/word-list?hl=en#letter-s 

**should, should be
Generally avoid.**
image

Comment (barriebyron) [reply docs/ibc/overview.md:182]: yes, much better @AdityaSripal (except "should" let's use "must" instead)

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:186]: yes better, the more I read this content, the move love it needs!

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:174]: oh man, now I went and rewrote a bit more for clarity

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:148]: is there a reason we don't use the packet definition from https://github.com/cosmos/ibc/blob/master/spec/core/ics-004-channel-and-packet-semantics/README.md 

(seems much more clear!)

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:151]: so the packet sequence that defines ordering is 
  `sequence: uint64` ?
  seems like we would want to define what is ORDERED and what is UNORDERED

File: docs/ibc/overview.md
Comment (barriebyron) [inline docs/ibc/overview.md:175]: since line 175 defines what happens for a packet timeout in an ORDERED channel, I attempted to describe what happens in an UNORDERED channel  on line 180

(we really gotta show this essential content some love... am I on the right track here?)

**Comment by barriebyron** [comment]
hi @crodriguezvega thank you so much for your attention. This essential content can still be improved, but I think we're getting closer. 

I'm here tomorrow Friday Nov 19 (Eastern time) but then I'm off all next week. Feel free to make changes as you need and merge in at will. 

@AdityaSripal you've been awesome with you input, thank you! @damiannolan thank you for weighing in on word choice.

Please give the IBC content the attention and care it requires in this PR, I don't want to hold up iterative improvements! We can use the netlify preview and have someone read it through for a fresh look, if that helps! 

(there is still some repetition)

File: docs/ibc/overview.md
Comment (AdityaSripal) [reply docs/ibc/overview.md:148]: I'm in favor of just copying that over.

The generic `Identifier` type in the spec should be changed to `string` for the go implementation

File: docs/ibc/overview.md
Comment (AdityaSripal) [reply docs/ibc/overview.md:151]: Yes if a channel is ORDERED, then the packets must be received in the order of the `sequence` that is inside the packet.

File: docs/ibc/overview.md
Comment (AdityaSripal) [inline docs/ibc/overview.md:170]:
```suggestion
If the timeout passes without the packet being successfully received, the packet can no longer be
received on the destination chain. The sending module can timeout the packet and take appropriate actions.
```

**Review by AdityaSripal — APPROVED** [review]
Since @barriebyron is on vacation next week, I suggest that she make final edits to this PR based on any remaining suggestions, and then we can merge this PR.

@barriebyron , if you have any remaining questions/suggestions that you cannot resolve on your own, then please leave them as comments on this PR and we can address them in a subsequent PR/issue.

Thanks for the help!

**Review by seantking — APPROVED** [review]

File: docs/ibc/overview.md
Comment (barriebyron) [reply docs/ibc/overview.md:148]: @AdityaSripal can you take on that spec work?

**Comment by barriebyron** [comment]
thanks everyone for the great collaboration! Please merge when tests complete!

**Pull request merged**
